src/utils/daemon.rs

```diff
@@ -11,20 +11,32 @@ use crate::{
 use chrono::{Timelike, Utc};
 use failure::Error;
 use log::{debug, error, info};
+use std::process::Command;
 use std::sync::Arc;
 use std::thread;
-use std::time::Duration;
+use std::time::{Duration, Instant};
+
+fn run_git_gc() {
+    Command::new("git")
+        .args(&["gc"])
```

Comment:
Can tweak the parameters here a little bit, adding [`--auto`](https://git-scm.com/docs/git-gc#Documentation/git-gc.txt---auto) will only actually gc when it's needed and we could maybe add a [`--prune=<date>`](https://git-scm.com/docs/git-gc#Documentation/git-gc.txt---pruneltdategt) since the repo has fairly regular updates
```suggestion
        .args(&["gc", "--auto"])
```
we *could* run [`--aggressive`](https://git-scm.com/docs/git-gc#Documentation/git-gc.txt---aggressive), but I'm not entirely sure if the cost to benefit ratio is worth it

Reply:
cc @jyn514 for the options of `--prune` and `--aggressive`

Reply:
Auto is a great idea. We definitely don't need aggressive since that will basically rewrite the whole index every time there's an update. Prune sounds interesting but since we only have 1 branch and no force pushes I don't think it's necessary.

Really this is just trying to avoid the worst case scenario of '3 years without git gc', it doesn't need to be super efficient.